crates/node/src/catchup.rs

```diff
@@ -35,8 +36,11 @@ impl Node {
         };
 
         let request = match from_json_slice(&message?.data)? {
-            CatchupStreamMessage::Request(req) => req,
+            CatchupStreamMessage::TransactionsRequest(req) => req,
             message @ (CatchupStreamMessage::TransactionsBatch(_)
+            | CatchupStreamMessage::ApplicationBlobRequest(_)
+            | CatchupStreamMessage::ApplicationBlobSize(_)
+            | CatchupStreamMessage::ApplicationBlobChunk(_)
```

Comment:
I wonder if it's possible to lose chunks or them arriving out of order, since this isn't necessarily TCP, in your stream research, did you come across any guarantees for data resilience?